enhancements/dns/externaldns-operator-aws-assume-role.md

```diff
@@ -0,0 +1,287 @@
+    // +required
+    ARN string `json:"arn,omitempty"`
+}
+
+```
+
+### Implementation Details/Notes/Constraints
+
+#### The Required Credentials Issue
+
+Extending the v1beta1 `ExternalDNS` API object revealed a pre-existing issue with the `Credentials` API field in
+`ExternalDNSAWSProviderOptions`. The problem is that `Credentials` is designated as `+required` to satisfy vanilla
+Kubernetes use cases. On OpenShift, the ExternalDNS Operator generates a `CredentialRequest` automatically, eliminating
+the need to manually specify `Credentials`. This wasn't yet exposed as an issue because there were no other fields in
+`spec.provider.aws`, and `spec.provider.aws` is optional. The underlying [logic](https://github.com/openshift/external-dns-operator/blob/11e3b72b75d696b7419bce8360183373fc0725e1/api/v1beta1/externaldns_webhook.go#L132)
+of the webhook is accurate; the issue lies with this required field in the `ExternalDNS` CRD.
+
+However, there exists a simple workaround: we can apply a default value of an empty string `""` to the
+`spec.aws.provider.credentials.name` field using the `+kubebuilder:default:={"name":""}` specifier on the `Credentials`
+field. When `spec.aws.provider` is created, but `spec.aws.provider.credentials` is not specified, it will default
+`spec.aws.provider.credentials.name` to `""`.
+
+This satisfies the `+required` field for `Credentials` while the ExternalDNS Operator logic [ignores](https://github.com/openshift/external-dns-operator/blob/bac092e98fe5f9065b75bd0fa21d5aeff9d00853/pkg/operator/controller/credentials-secret/controller.go#L259-L267)
+the empty string and uses the `CredentialsRequest`. Conversely, vanilla Kubernetes users (who do not have the option to
+use a `CredentialsRequest`) will encounter a validation webhook failure when the `Credentials` default of `""` is used
+due to the [validateProviderCredentials](https://github.com/openshift/external-dns-operator/blob/4cb22f19c5e6edcc6a37b73e17f25d996dbef9cd/api/v1beta1/externaldns_webhook.go#L132-L160). This is desirable because non-OpenShift users must specify valid `Credentials`.
+This failure is desirable because vanilla Kubernetes users are required to specify valid `Credentials`.
+
+In a future version of ExternalDNS Operator's API, we will resolve the requirement for the `Credentials` field through a
+more appropriate solution as outlined in the alternative [Bumping the API](#bumping-the-api).
+
+### Risks and Mitigations
+
+N/A
+
+### Drawbacks
+
+A minor drawback is the added complexity in supporting cross-account DNS hosted zones. This complexity involves handling
+AWS resource sharing, managing IAM policies and roles, and understanding shared VPC architecture.
+
+In terms of security concerns, the IAM role is not considered sensitive because AWS incorporates security measures to
+limit the usage of a role ARN to a particular AWS account. Exploitation of the `AssumeRole` field isn't a concern due to
+these AWS security constraints.
```

Comment:
I'm still a bit unconvinced that adding this functionality doesn't introduce new security concerns, but I can let it go.

Reply:
I suppose it's worded too strongly, I'll remove.